runtime-sdk/src/storage/typed.rs

```diff
@@ -57,17 +69,15 @@ impl<'store, K: DecodableStoreKey, V: DeserializeOwned> TypedStoreIterator<'stor
     }
 }
 
-impl<'store, K: DecodableStoreKey, V: DeserializeOwned> Iterator
-    for TypedStoreIterator<'store, K, V>
+impl<'store, K, V> Iterator for TypedStoreIterator<'store, K, V>
+where
+    K: for<'k> TryFrom<&'k [u8]>,
+    V: DeserializeOwned,
 {
-    type Item = (K, V);
+    type Item = (K, V); // TODO: Should this return `Result`?
```

Comment:
This would make it more awkward for the caller to use. And if deserialization fails, this means that the stored state is somehow corrupted/incompatible, so I'm not sure if we should let the caller handle it? Are there any nice ways of handling such iterators in the caller, by e.g. propagating an error there if needed without requiring too much boilerplate? In case its not too bad we could change it to `Result` here.

Currently it seems you will just return `None` which will silently hide the corruption/incompatibility which is definitely not what we want. If the type is not `Result` this should panic instead (which will crash the runtime and result in the compute node voting "fail" for the batch).